compiler_gym/envs/mlir/service/MlirUtils.cc

```diff
@@ -0,0 +1,285 @@
+    vectorContractLowering = mlir::vector::VectorContractLowering::Dot;
+  } else if (vectorizeToAction.int64_value() == 1) {
+    vectorContractLowering = mlir::vector::VectorContractLowering::Matmul;
+  } else if (vectorizeToAction.int64_value() == 2) {
+    vectorContractLowering = mlir::vector::VectorContractLowering::OuterProduct;
+  } else {
+    CHECK(false) << fmt::format("Unexpected vectorize_to \"{}\".", vectorizeToAction.int64_value());
+  }
+
+  if (vectorizeTransferSplitAction.int64_value() == 0) {
+    vectorTransferSplit = mlir::vector::VectorTransferSplit::None;
+  } else if (vectorizeTransferSplitAction.int64_value() == 1) {
+    vectorTransferSplit = mlir::vector::VectorTransferSplit::LinalgCopy;
+  } else if (vectorizeTransferSplitAction.int64_value() == 2) {
+    vectorTransferSplit = mlir::vector::VectorTransferSplit::VectorTransfer;
+  } else {
+    CHECK(false) << fmt::format("Unexpected vector_transfer_split \"{}\".",
+                                vectorizeTransferSplitAction.int64_value());
+  }
+
+  unrollVectorTransfers = unrollVectorTransfersAction.boolean_value();
+}
+
+void LinalgCodegenPass::runStrategy(const Event& action, StringRef anchorOpName) {
+  mlir::linalg::CodegenStrategy strategy;
+  mlir::linalg::LinalgTilingOptions tilingOptions;
+  mlir::vector::VectorContractLowering vectorContractLowering;
+  mlir::vector::VectorTransferSplit vectorTransferSplit;
+  bool unrollVectorTransfers, promote, promote_full_tile;
+
+  performTileOptions(action, tilingOptions, promote, promote_full_tile);
+  performVectorizeOptions(action, vectorContractLowering, vectorTransferSplit,
+                          unrollVectorTransfers);
+
+  strategy.tile(anchorOpName, tilingOptions)
+      .promoteIf(
+          promote, anchorOpName,
+          mlir::linalg::LinalgPromotionOptions().setAlignment(16).setUseFullTileBuffersByDefault(
+              promote_full_tile))
+      .vectorize(anchorOpName)
+      .vectorLowering(
+          mlir::linalg::LinalgVectorLoweringOptions()
+              .setVectorTransformsOptions(mlir::vector::VectorTransformsOptions()
+                                              .setVectorTransformsOptions(vectorContractLowering)
+                                              .setVectorTransferSplit(vectorTransferSplit))
+              .setVectorTransferToSCFOptions(
+                  mlir::VectorTransferToSCFOptions().enableFullUnroll(unrollVectorTransfers))
+              .enableTransferPartialRewrite()
+              .enableContractionLowering()
+              .enableTransferToSCFConversion());
+  // Created a nested OpPassManager and run.
+  mlir::FuncOp funcOp = getOperation();
+  mlir::OpPassManager dynamicPM("builtin.func");
+  strategy.configurePassPipeline(dynamicPM, funcOp.getContext());
+  if (failed(runPipeline(dynamicPM, funcOp)))
+    return signalPassFailure();
+}
+
+void LinalgCodegenPass::runOnOperation() {
+  // TODO(kyleherndon): figure out why this won't compile/remove it
```

Comment:
Should this be done before merging?

Reply:
These parameters are now part of the input MLIR file. I removed this.